youtube_dl/extractor/vlive.py

```diff
@@ -31,16 +34,73 @@ def _real_extract(self, url):
 
-        long_video_id = self._search_regex(
-            r'vlive\.tv\.video\.ajax\.request\.handler\.init\(\s*"[0-9]+"\s*,\s*"[^"]*"\s*,\s*"([^"]+)"',
-            webpage, 'long video id')
+        status = self._download_json(
+            'http://www.vlive.tv/video/status?videoSeq=%s' % video_id,
+            video_id, 'Downloading JSON status',
+            headers={'Referer': url}).get('status')
 
-        key = self._search_regex(
-            r'vlive\.tv\.video\.ajax\.request\.handler\.init\(\s*"[0-9]+"\s*,\s*"[^"]*"\s*,\s*"[^"]+"\s*,\s*"([^"]+)"',
-            webpage, 'key')
+        video_params = self._search_regex(
+            r'vlive\.tv\.video\.ajax\.request\.handler\.init\((.*)\)',
+            webpage, 'video params')
+        live, long_video_id, key = re.split(r'"\s*,\s*"', video_params)[1:4]
 
+        if status == 'LIVE_ON_AIR' or status == 'BIG_EVENT_ON_AIR':
+            live_params = json.loads(json.loads('"%s"' % live))
+            return self._live(video_id, webpage, live_params)
+        elif status == 'VOD_ON_AIR' or status == 'BIG_EVENT_INTRO':
+            return self._replay(video_id, webpage, long_video_id, key)
+        elif status == 'LIVE_END':
+            raise ExtractorError('Uploading for replay. Please wait…',
+                                 expected=True)
+        elif status == 'COMING_SOON':
+            raise ExtractorError('Coming soon!', expected=True)
+        else:
+            raise ExtractorError('Unknown status %s' % status)
+
+    def _get_common_fields(self, webpage):
         title = self._og_search_title(webpage)
+        creator = self._html_search_regex(
+            r'<div[^>]+class="info_area"[^>]*>\s*<a\s+[^>]*>([^<]+)',
+            webpage, 'creator', fatal=False)
+        thumbnail = self._og_search_thumbnail(webpage)
+        return {
+            'title': title,
+            'creator': creator,
+            'thumbnail': thumbnail,
+        }
+
+    def _get_sub_url(self, video_url, lang):
+        url = video_url.split('/playlist.m3u8')[0]
+        url += '/subtitlelist_l%s.m3u8' % lang
+        return url
+
+    def _live(self, video_id, webpage, live_params):
+        formats = [{
+            'url': vid['cdnUrl'],
+            'ext': 'mp4',
+            'format_id': vid.get('name'),
+            'quality': int_or_none(vid.get('name')),
+            'width': int_or_none(vid.get('width')),
+            'height': int_or_none(vid.get('height')),
+        } for vid in live_params.get('resolutions', [])]
+        self._sort_formats(formats)
+
+        subtitles = {
+            'eng': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'eng')}
+                    for f in formats],
+            'zcn': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'zcn')}
+                    for f in formats],
+            'vie': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'vie')}
+                    for f in formats],
+        }
```

Comment:
Parsing should be the way. It's better than hard-coded values in general. An approach is modifying [```_extract_m3u8_formats```](https://github.com/rg3/youtube-dl/blob/d671237/youtube_dl/extractor/common.py#L1061-L1179) so that it can handle subtitles, too.